docs/source/guides/codal-projects.rst

```diff
@@ -28,6 +28,59 @@ ready to compile.
+Debugging is an important part of software development, as it allows you to find errors which might be hiding within your code that are tricky to
+spot otherwise. It allows you to step through your code, which means you can look at the data and memory of your program after each individual line
+of code is ran, as well as to set breakpoints, which tell your code to pause once it reaches a certain line of code, until you hit resume manually.
+
+You can find out more details about debugging within Visual Studio Code on their `website <https://code.visualstudio.com/docs/editor/debugging>`_.
+
+To set this up, we need to install a debugger for your hardware. This is a program which allows Visual Studio Code to connect through to your
+physical device (e.g. micro:bit) to send and receive the data it needs to operate. Download the ``Cortex-Debug`` extension for Visual Studio Code.
+This is a debugger which provides support for the type of devices which run CODAL.
+
+Next, make sure you have downloaded and installed either pyOCD or OpenOCD, as instructed within :doc:`/guides/prerequisites`.
+
+Then, you need to `download this configuration file <https://raw.githubusercontent.com/lancaster-university/microbit-v2-samples/master/.vscode/launch.json>`.
+This is a file which sets up debugging configurations within Visual Studio Code. This file is specifically designed for debugging micro:bit v2 devices,
+and so therefore you will need to modify this file as needed if you are using a different device. Consult the documentation for your specific device
+if this is the case.
+
+Place the file you have downloaded into the ``.vscode`` folder inside your application folder. If this folder doesn't exist, you should manually create it.
+
+At this point, you can switch to the ``Run and Debug`` panel, on the left. You should see a drop-down box at the top of the panel, containing
+``micro:bit PyOCD Cortex Debug`` and ``micro:bit OpenOCD Cortex Debug``. These are the configurations you installed by downloading the ``launch.json`` file.
+You should select the appropriate option depending on which debugger you installed.
+
+Next, ensure your micro:bit is plugged into your computer, and hit the green arrow to start debugging! This will make sure your program is compiled and
+copied to your micro:bit, meaning you can skip the process of running the ``build.py`` script manually.
+
+To test out the debugger, try placing a breakpoint. You can do this by clicking in the space to the left of the corresponding line number in the code editor.
+Restart your program, and see if your program pauses at the breakpoints you set!
+
+.. note::
+    Unfortunately, sometimes due to a variety of reasons, attempting to start the debugger can result in errors. Much of the time this can be resolved by
+    trying again, reconnecting your device to your computer, or restarting your computer entirely.
```

Comment:
Yeah, this isn't the greatest situation to be in, and we should definitely follow up on figuring out why in _some cases_ this fails...